src/rating.tsx

```diff
@@ -0,0 +1,349 @@
+import IconFaceNeutralFilled from './generated/mistica-icons/icon-face-neutral-filled';
+import IconFaceNeutralRegular from './generated/mistica-icons/icon-face-neutral-regular';
+import IconFaceHappyFilled from './generated/mistica-icons/icon-face-happy-filled';
+import IconFaceHappyRegular from './generated/mistica-icons/icon-face-happy-regular';
+import IconFaceSuperHappyFilled from './generated/mistica-icons/icon-face-super-happy-filled';
+import IconFaceSuperHappyRegular from './generated/mistica-icons/icon-face-super-happy-regular';
+import {useTheme} from './hooks';
+
+import type {ExclusifyUnion} from './utils/utility-types';
+import type {DataAttributes, IconProps} from './utils/types';
+
+type RatingIconProps = {
+    ActiveIcon: React.FC<IconProps>;
+    InactiveIcon: React.FC<IconProps>;
+    color: string;
+};
+
+const DEFAULT_RATING_SIZE = 32;
+const DEFAULT_INFO_RATING_SIZE = 16;
+const DEFAULT_RATING_ICON_COUNT = 5;
+
+const DEFAULT_RATING_ICON: RatingIconProps = {
+    ActiveIcon: IconStarFilled,
+    InactiveIcon: IconStarRegular,
+    color: vars.colors.controlActivated,
+};
+
+const DEFAULT_INFO_RATING_ICON: RatingIconProps = {
+    ActiveIcon: IconStarFilled,
+    InactiveIcon: IconStarRegular,
+    color: vars.colors.warning,
+};
+
+const DEFAULT_QUALITATIVE_ICONS: Array<RatingIconProps> = [
+    {
+        ActiveIcon: IconFaceSadFilled,
+        InactiveIcon: IconFaceSadRegular,
+        color: vars.colors.errorHigh,
+    },
+    {
+        ActiveIcon: IconFaceSlightlySadFilled,
+        InactiveIcon: IconFaceSlightlySadRegular,
+        color: vars.colors.error,
+    },
+    {
+        ActiveIcon: IconFaceNeutralFilled,
+        InactiveIcon: IconFaceNeutralRegular,
+        color: vars.colors.warning,
+    },
+    {
+        ActiveIcon: IconFaceHappyFilled,
+        InactiveIcon: IconFaceHappyRegular,
+        color: vars.colors.success,
+    },
+    {
+        ActiveIcon: IconFaceSuperHappyFilled,
+        InactiveIcon: IconFaceSuperHappyRegular,
+        color: vars.colors.successHigh,
+    },
+];
+
+interface BaseRatingProps {
+    size?: number;
+    dataAttributes?: DataAttributes;
+    valueLabels?: Array<string>;
+    'aria-label'?: string;
+    'aria-labelledby'?: string;
+}
+
+interface QuantitativeRatingProps extends BaseRatingProps {
+    type?: 'quantitative';
+    icon?: RatingIconProps;
+    count?: number;
+    icons?: never;
+}
+
+interface QualitativeRatingProps extends BaseRatingProps {
+    type: 'qualitative';
+    icons?: Array<RatingIconProps>;
+    icon?: never;
+    count?: never;
+}
+
+type RatingProps = ExclusifyUnion<QualitativeRatingProps | QuantitativeRatingProps> & {
+    value?: number;
+    defaultValue?: number;
+    onChangeValue?: (value: number) => void;
+    disabled?: boolean;
+};
+
+type InfoRatingProps = Omit<QuantitativeRatingProps, 'type' | 'valueLabels'> & {
+    value?: number;
+    withHalfValue?: boolean;
```

Comment:
what does this mean? is this to be able to control the precision (by half star or whole star)? why isn't this always true for InfoRating (and remove the prop)? If someone doesn't want half star precision they can Math.round from outside

Reply:
Because of design specs:

![image](https://github.com/user-attachments/assets/50d65839-a1f9-4d31-a49d-787c7640b7ed)

Reply:
this sounds weird to me